src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -302,6 +302,400 @@ bool Pane::NavigateFocus(const FocusDirection& direction)
     return false;
 }
 
+// Method Description:
+// - Attempts to swap the location of the two given panes in the tree.
+//   Searches the tree starting at this pane to find the parent pane for each of
+//   the arguments, and if both parents are found, replaces the appropriate
+//   child in each.
+// Arguments:
+// - first: A pointer to the first pane to switch.
+// - second: A pointer to the second pane to switch.
+// Return Value:
+// - true if a swap was performed.
+bool Pane::SwapPanes(std::shared_ptr<Pane> first, std::shared_ptr<Pane> second)
+{
+    // If there is nothing to swap, just return.
+    if (first == second || _IsLeaf())
+    {
+        return false;
+    }
+
+    std::unique_lock lock{ _createCloseLock };
+
+    // Recurse through the tree to find the parent panes of each pane that is
+    // being swapped.
+    std::vector<std::shared_ptr<Pane>> parents;
+    parents.push_back(shared_from_this());
+    std::shared_ptr<Pane> firstParent = nullptr;
+    std::shared_ptr<Pane> secondParent = nullptr;

```

Comment:
I did it this way to avoid having yet another recursive helper function, but also it is somewhat vestigial on when I captured additional state and tried to modify it in place. I can see about refactoring a bit.

Reply:
I created a `_FindParentOfPane` function and used it to replace the the vector-based logic.